lms/static/scripts/frontend_apps/components/dashboard/DashboardApp.tsx

```diff
@@ -1,29 +1,65 @@
+import {
+  Link,
+  LogoIcon,
+  ProfileIcon,
+  SettingsIcon,
+} from '@hypothesis/frontend-shared';
 import classnames from 'classnames';
-import { Route, Switch, useParams } from 'wouter-preact';
+import { Link as RouterLink, Route, Switch, useParams } from 'wouter-preact';
 
+import { useConfig } from '../../config';
 import AssignmentActivity from './AssignmentActivity';
 import CourseActivity from './CourseActivity';
 import DashboardFooter from './DashboardFooter';
 import OrganizationActivity from './OrganizationActivity';
 
 export default function DashboardApp() {
+  const { dashboard } = useConfig(['dashboard']);
   const { organizationPublicId } = useParams<{
     organizationPublicId: string;
   }>();
 
   return (
     <div className="flex flex-col min-h-screen gap-5">
-      <div
-        className={classnames(
-          'flex justify-center p-3 w-full',
-          'bg-white border-b shadow',
-        )}
-      >
-        <img
-          alt="Hypothesis logo"
-          src="/static/images/hypothesis-wordmark-logo.png"
-          className="h-10"
-        />
+      <div className="px-3 py-4 bg-white border-b shadow">
+        <div className="flex justify-between items-center mx-auto max-w-6xl">
+          <RouterLink href="" aria-label="All courses">
+            <LogoIcon />
+          </RouterLink>
+
+          <div className="flex gap-6">
+            <span
+              className={classnames(
+                'flex gap-2 items-center',
+                'font-semibold text-color-text-light',
+              )}
+              data-testid="display-name"
+            >
+              <ProfileIcon />
+              {dashboard.user.display_name}
+            </span>
+            <Link
+              underline="hover"
+              variant="text"
+              href="/email/preferences"
+              target="_blank"
+              classes={classnames(
+                'flex gap-2 items-center',
+                'font-semibold text-color-text-light',
+              )}
+              onClick={
+                /* istanbul ignore next - Temporary until settings link works */
+                e => {
+                  e.preventDefault();
+                  alert('This is not implemented yet');
```

Comment:
I was going to use `confirm` from frontend-shared, as a slightly nicer temp approach, but it enforces two buttons for "accept" and "cancel", so I just used a regular `alert` for now.